llvm/tools/llvm-c-test/debuginfo.c

```diff
@@ -224,14 +235,46 @@ int llvm_test_dibuilder(void) {
   LLVMPositionBuilderBeforeInstrAndDbgRecords(Builder, InsertPos);
   LLVMValueRef Phi1 = LLVMBuildPhi(Builder, I64, "p1");
   LLVMAddIncoming(Phi1, &Zero, &FooEntryBlock, 1);
+
   // Do the same again using the other position-setting function.
   LLVMPositionBuilderBeforeDbgRecords(Builder, FooVarBlock, InsertPos);
   LLVMValueRef Phi2 = LLVMBuildPhi(Builder, I64, "p2");
   LLVMAddIncoming(Phi2, &Zero, &FooEntryBlock, 1);
+
   // Insert a non-phi before the `ret` but not before the debug records to
   // test that works as expected.
   LLVMPositionBuilder(Builder, FooVarBlock, Ret);
-  LLVMBuildAdd(Builder, Phi1, Phi2, "a");
+  LLVMValueRef Add = LLVMBuildAdd(Builder, Phi1, Phi2, "a");
+
+  // Iterate over debug records in the add instruction. There should be two.
+  // Iterate with a for loop.
+  LLVMDbgRecordRef DbgRec;
+  for (DbgRec = LLVMGetFirstDbgRecord(Add); DbgRec;
+       DbgRec = LLVMGetNextDbgRecord(DbgRec)) {
+    assert(DbgRec != NULL);
```

Comment:
hmm given the for-loop's condition is that `DbgRec` is non-null, this assert feels redundant. I think all the functions used in these loops are used in the "manual iteration" below, so we can probably just cut these two loops from the test?

Reply:
Done

Reply:
Not sure if I'm looking at an old diff or something but I still see the for loops

Reply:
Now should be pushed, sorry